common/RELEASING.md

```diff
@@ -0,0 +1,25 @@
+# Releasing
+
+Set variables:
+
+    $ export CRATE=$(cat Cargo.toml | grep name | sed 's/.*"\([^"]*\)".*/\1/')
+    $ export VERSION=X.Y.Z
+    $ export GPG_KEY=F001234
+
+Update version numbers:
+
+    $ vim Cargo.toml
+
+Update changelog:
+
+    $ vim CHANGELOG.md
+
+Commit & tag:
+
+    $ git commit -S${GPG_KEY} -m "${CRATE}: Release v${VERSION}"
+    $ git tag -s -u ${GPG_KEY} ${CRATE}-v${VERSION} -m "${CRATE}: Version ${VERSION}"
```

Comment:
I consider tagging an un-pushed commit which didn't go through CI bad.

Reply:
So I'd to all the changes in a prepare-release branch, tag the merge commit once it has gone through CI and review and is merged. Then when the tag is pushed run cargo publish.

Ideally the CI would even do the cargo publish after one pushed the tag.

Reply:
> I consider tagging an un-pushed commit which didn't go through CI bad.

That's why the push happens only after "cargo publish", which runs the testsuite prior to publishing.

> So I'd do all the changes in a prepare-release branch, tag the merge commit once it has gone through CI and review and is merged. Then when the tag is pushed run cargo publish.

Would be fine for me, but note that right now there is no CI set up at all. And if we set up CI, that will need to test all PACs, so even if another PAC fails the CI checks, publishing would be blocked.

Reply:
> That's why the push happens only after "cargo publish", which runs the testsuite prior to publishing.

CI is not only about running the testsuite. It' may also include clippy, formatting and other stuff. Also reviewing prior to a release may also be important.